### PR TITLE
How-To for Reflection and Hairpin NAT added

### PR DESCRIPTION
**Pull request by Monviech**
I added a how-to nat_reflection.rst for Reflection and Hairpin NAT and tried to embed is as naturally as possible into the existing documentation. I also changed nat.rst to refer to the how-to section and improved some of the Reflection descriptions in firewall_settings.rst

I'm open to discussion and I'm sure this will take a while to review because it's a big addition.

### CODE REVIEW

File: source/manual/firewall_settings.rst [hunk 1]
```diff
@@ -21,42 +21,47 @@ Network Address Translation
...
     very explicit when one inspects your setup.
 
+.. Tip::
+   There is a how-to section explaining :doc:`NAT Reflection <how-tos/nat_reflection>` in detail.
+    
+.. Attention::
+    Firewall Rules won't be automatically generated when using any of the below Reflection options. You have to create them manually or traffic will be blocked by the default deny rule.
+    
+.. Note::
+    * Automatic Reflection rules aren't visible in the GUI. Examine them with ``pfctl -s nat`` in the shell.
```
Comment (AdSchellevis) [inline source/manual/firewall_settings.rst:36]: or "Firewall: Diagnostics: Statistics => rules" for a raw view in the gui

**Review by AdSchellevis — APPROVED** [review]

**Comment by Monviech** [comment]
@AdSchellevis I think the tutorial is in a good shape, I have let some people test it the past weeks and the general feedback was that it resolved their NAT problems.

Two examples:
https://forum.opnsense.org/index.php?topic=36241.0
https://forum.opnsense.org/index.php?topic=36071

Would it be best practice to add my real name as author when I create sections in the doc?

**Comment by AdSchellevis** [comment]
@Monviech for the docs we don't really have options to annotate the copyright (although one or two how-to's contain a contributor name), I can ask @fichtner to mention you in the release notes of the product.

**Comment by Monviech** [comment]
@AdSchellevis Sounds good. I'll continue to write tutorials if I find some things that can be improved. :)

**Comment by AdSchellevis** [comment]
@Monviech ok, just let @fichtner known which name to add and I'll take a look at this PR (and merge it asap). Thanks for your contribution!

**Comment by AdSchellevis** [comment]
@Monviech I just took the time to merge and compile the docs, one small issue with an underline, but the text is really good, thanks a lot, these efforts are much appreciated, this is a very good outline of the problem and implementation scenario's.

**Comment by Monviech** [comment]
@AdSchellevis I appreciate your feedback. I'm happy you like my style of writing. The compiling issue went unnoticed by me because I use sphinx-autobuild while editing the docs so it's not always compiled fully when I edit something. I also have a lot f compile errors due to using python3 it seems.

**Pull request merged**